Functions/Ttry.py

```diff
@@ -1,4 +1,47 @@
 from Definitions.types import TYPE
 
+# Input example
+# node = {
+#    "description": "ttry",
+#    "elements": {
+#        "e1": {
+#            "description": "traise"
+#            "elements": {
+#                "e1": "raise"
+#            }
+#        },
+#        "e2": {
+#            "description": "tbool"
+#            "elements": {
+#                "e1": "false"
+#            }
+#        }
+#   }
+# }
+# Expected Output
+# TYPE.BOOL
+
+def term_is_valid(term):
+
+    return term and "elements" in term and "e1" in term["elements"] and "e2" in term["elements"]
+
 def t_try(environment, term):
+
+    from main import infer_type
+
+    if not term_is_valid(term):
+        return TYPE.ERROR
+
+    type_e1 = infer_type(environment, term["elements"]["e1"])
+    type_e2 = infer_type(environment, term["elements"]["e2"])
+
+    if type_e1 == type_e2:
```

Comment:
Isso aqui vai dar erro se os 2 forem raise raise
Eu não tenho certeza se em algum universo isso faz sentido.

Mas como tá agora um programa `raise raise` vai retornar `TYPE.UNDEFINED`
Vê se isso faz sentido

Reply:
Se for try raise with raise, o resultado esperado pra essa expressão é UNDEFINED. Então tá certo retornar tanto type_e1 ou type_e2, já que os dois são iguais e iguais a UNDEFINED